### PR TITLE
Discover page

### LINKED ISSUE #15: Discover/Search screen
# As a dev,
I want to have a Search screen, 
so that the user is able to find single shows.

# Description
Create basic layout for the Screen with Header and Search Component.

# Done?
- [x] Add Header component
- [x] Add Screen layout with basic styles
- [x] Add Input / Search component

### PR DESCRIPTION
**Pull request by mschmauder-hub**
Fix #15 

This is just the basic structure of my Search screen. I have added a Header and Input component. Once, I have finished my other components for this screen I  will work on the styling. I split this screen up into 2 pull requests, because there are still some other small adjustments I need to make and it would make this pull request a bit bloated and hard to understand.  I also had to refactor my file structure, so it will be easier to organize my files from now on.

Header_and_SearchBar

### CODE REVIEW

File: SeriesMaster-Frontend/App/Screens/DiscoverScreen.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    <Container>
+      <Header title="Discover"></Header>
+      <Main>
+        <AppInputText
+          placeholder="Search"
+          query={query}
+          onChange={handleChange}
+        />
+      </Main>
+      <View style={{ padding: 10 }}>
```
Comment (lmachens) [inline SeriesMaster-Frontend/App/Screens/DiscoverScreen.js:30]: You mix inline styles and CSS classes.
Better move this `padding` to a styled component like you did for the `Container`.

File: SeriesMaster-Frontend/App/Screens/DiscoverScreen.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  }
+
+  return (
+    <Container>
+      <Header title="Discover"></Header>
+      <Main>
+        <AppInputText
+          placeholder="Search"
+          query={query}
+          onChange={handleChange}
```
Comment (lmachens) [inline SeriesMaster-Frontend/App/Screens/DiscoverScreen.js]: You don't need a `handleChange` if you forward all props to another function.

```suggestion
          onChange={setQuery}
```

File: SeriesMaster-Frontend/App/components/AppInputText.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  padding: 10px;
+  margin-left: 5px;
+`;
+
+const AppInputText = ({ placeholder, query, onChange }) => {
+  return (
+    <InputContainer>
+      <InputText
+        value={query}
+        onChange={() => onChange(event.target.value)}
```
Comment (lmachens) [inline SeriesMaster-Frontend/App/components/AppInputText.js:22]: `event` is undefined

File: SeriesMaster-Frontend/App/Screens/DiscoverScreen.js [hunk 4]
```diff
@@ -0,0 +1,45 @@
+import React, { useState } from "react";
+import styled from "styled-components/native";
+import { StatusBar, View, Text } from "react-native";
+import Header from "../components/Header";
+import AppInputText from "../components/AppInputText";
+
+const Container = styled.View`
+  flex: 1;
```
Comment (marwinburesch) [inline SeriesMaster-Frontend/App/Screens/DiscoverScreen.js:8]: I don't see the corresponding parent element, which needs `display: flex` to make this line viable

File: SeriesMaster-Frontend/App/components/AppInputText.js [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+        placeholder={placeholder}
+      />
+    </InputContainer>
+  );
+};
+
+AppInputText.propTypes = {
+  query: PropTypes.any,
+  onChange: PropTypes.any,
```
Comment (lmachens) [inline SeriesMaster-Frontend/App/components/AppInputText.js]:
```suggestion
  query: PropTypes.string,
  onChange: PropTypes.func,
```

File: SeriesMaster-Frontend/App/components/Header.js [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+      <StyledText>{title}</StyledText>
+      <StyledImage source={require("../assets/menu-white.svg")} />
+    </HeaderStyle>
+  );
+};
+
+export default Header;
+
+Header.propTypes = {
+  title: PropTypes.any,
```
Comment (lmachens) [inline SeriesMaster-Frontend/App/components/Header.js]:
```suggestion
  title: PropTypes.string,
```

File: SeriesMaster-Frontend/App/components/Header.js [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+const StyledImage = styled.Image`
+  height: 30px;
+  width: 30px;
+`;
+
+const Header = ({ title }) => {
+  return (
+    <HeaderStyle>
+      <StyledText>{title}</StyledText>
+      <StyledImage source={require("../assets/menu-white.svg")} />
```
Comment (lmachens) [inline SeriesMaster-Frontend/App/components/Header.js]: You mix CommonJS and Modules.
Instead, import the SVG on top of this file.

File: SeriesMaster-Frontend/App/Screens/DiscoverScreen.js [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+  align-items: center;
```
Comment (marwinburesch) [inline SeriesMaster-Frontend/App/Screens/DiscoverScreen.js:9]: So `View` is already a flexbox?

**Review by lmachens — CHANGES_REQUESTED** [review]
Good job 👍 ! Only small issues.

The `225030.jpg` file name is unclear. I guess, this is only temporary?

In addition, link this pull request with your project board.

File: SeriesMaster-Frontend/App/components/Header.js [hunk 9]
```diff
@@ -0,0 +1,41 @@
+import React from "react";
+import styled from "styled-components/native";
+import PropTypes from "prop-types";
+
+const HeaderStyle = styled.View`
+  display: flex;
```
Comment (marwinburesch) [inline SeriesMaster-Frontend/App/components/Header.js]: So View is not already a flexbox? xD

File: SeriesMaster-Frontend/App/Screens/DiscoverScreen.js (hunk 1 above)
Comment (marwinburesch) [inline SeriesMaster-Frontend/App/Screens/DiscoverScreen.js:30]: From what I've seen so far, using inline styles seems unconventional 🧐

**Review by marwinburesch — CHANGES_REQUESTED** [review]
Good stuff, mate! That React Native View is confusing :D :D

cheers! 🐜

File: SeriesMaster-Frontend/App/components/AppInputText.js (hunk 3 above)
Comment (mschmauder-hub) [reply SeriesMaster-Frontend/App/components/AppInputText.js:22]: In this case it doesn't seem to make a difference. I can still log the event and it is recognizing it as an input event.

File: SeriesMaster-Frontend/App/components/Header.js (hunk 7 above)
Comment (mschmauder-hub) [reply SeriesMaster-Frontend/App/components/Header.js]: I didn't think of that. Thanks!

File: SeriesMaster-Frontend/App/components/Header.js (hunk 9 above)
Comment (mschmauder-hub) [reply SeriesMaster-Frontend/App/components/Header.js]: Thanks, I just removed that line. xD

File: SeriesMaster-Frontend/App/Screens/DiscoverScreen.js (hunk 1 above)
Comment (mschmauder-hub) [reply SeriesMaster-Frontend/App/Screens/DiscoverScreen.js:30]: I used it as a placeholder until I build the component. But yes I should have used a Styled component instead.

**Pull request merged**